core/unix/os.c

```diff
@@ -8178,6 +8178,8 @@ post_system_call(dcontext_t *dcontext)
         /* We assumed in pre_system_call() that the unmap would succeed
          * and flushed fragments and removed the region from exec areas.
          * If the unmap failed, we re-add the region to exec areas.
+         * For zero-lenght unmaps we don't need to re-add anything,
```

Comment:
spelling: length

Reply:
It let me fix it so I went ahead and did that and will merge after the tests run.
Thanks again for the patch!